src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -55,70 +58,175 @@ static void check(Coordination::Error code, const std::string & path)
         throw KeeperException::fromPath(code, path);
 }
 
+UInt64 getSecondsUntilReconnect(const ZooKeeperArgs & args)
+{
+    std::uniform_int_distribution<UInt32> fallback_session_lifetime_distribution
+        {
+            args.fallback_session_lifetime.min_sec,
+            args.fallback_session_lifetime.max_sec,
+        };
+    UInt32 session_lifetime_seconds = fallback_session_lifetime_distribution(thread_local_rng);
+    return session_lifetime_seconds;
+}
+
+Coordination::ZooKeeper::Node hostToNode(const LoggerPtr & log, const ShuffleHost & host)
+{
+    /// We want to resolve all hosts without DNS cache for keeper connection.
+    Coordination::DNSResolver::instance().removeHostFromCache(host.host);
```

Comment:
Do not understand how it helps here.
If we never resolve keeper node with `DNSResolver` then there is nothing to delete.
Otherwise this only drop the cache only when it is called, it still might be involved between such drops. 

This line
`const Poco::Net::SocketAddress host_socket_addr{host.host};` 
do not use `DNSResolver` for resolving hostname.

Reply:
This change was introduced in https://github.com/ClickHouse/ClickHouse/pull/50738
We can ask @pufit

Reply:
Yes, it's redundant now

Reply:
Yes, I forgot how this PR was merged, but it seems unfinished to me 😅